src/main/java/com/coreos/jetcd/resolver/DnsSrvNameResolver.java

```diff
@@ -0,0 +1,81 @@
+package com.coreos.jetcd.resolver;
+
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.List;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.Attributes;
+import io.grpc.ResolvedServerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class DnsSrvNameResolver extends AbstractEtcdNameResolver {
+    private final static Logger LOGGER;
+    private final static String[] ATTRIBUTE_IDS;
+    private final static Hashtable<String, String> ENV;
+
+    static {
+        LOGGER = LoggerFactory.getLogger(DnsSrvNameResolver.class);
+        ATTRIBUTE_IDS = new String[] { "SRV" };
+
+        ENV = new Hashtable<>();
+        ENV.put("java.naming.factory.initial", "com.sun.jndi.dns.DnsContextFactory");
+        ENV.put("java.naming.provider.url", "dns:");
+    }
+
+    private final String name;
+
+    public DnsSrvNameResolver(URI uri) {
+        super(uri);
+
+        if (!uri.getAuthority().startsWith("_etcd-server._tcp.")) {
+            this.name = "_etcd-server._tcp." + getServiceAuthority();
+        } else {
+            this.name = getServiceAuthority();
+        }
+    }
+
+    @VisibleForTesting
+    DnsSrvNameResolver(String targetUri) {
+        super(URI.create(targetUri));
+        this.name = getServiceAuthority();
+    }
+
+    @Override
+    protected List<ResolvedServerInfo> resolve() {
+        try {
```

Comment:
I think it's better to move logical here to a separate thread. Even in `SimpleEtcdNameResolver` I want to use a separate thread to update listener.